generated/nixnet/nixnet_service.cpp

```diff
@@ -370,6 +373,39 @@ namespace nixnet_grpc {
     }
   }
 
+  //---------------------------------------------------------------------
+  //---------------------------------------------------------------------
+  ::grpc::Status NiXnetService::DbCreateObject(::grpc::ServerContext* context, const DbCreateObjectRequest* request, DbCreateObjectResponse* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      auto parent_object_ref_grpc_session = request->parent_object_ref();
+      nxDatabaseRef_t parent_object_ref = nx_database_ref_t_resource_repository_->access_session(parent_object_ref_grpc_session.id(), parent_object_ref_grpc_session.name());
+      u32 object_class = request->object_class();
+      auto object_name = request->object_name().c_str();
+
+      auto init_lambda = [&] () {
+        nxDatabaseRef_t db_object_ref;
+        auto status = library_->DbCreateObject(parent_object_ref, object_class, object_name, &db_object_ref);
+        return std::make_tuple(status, db_object_ref);
+      };
+      uint32_t session_id = 0;
+      const std::string& grpc_device_session_name = request->session_name();
+      auto cleanup_lambda = [&] (nxDatabaseRef_t id) { library_->DbDeleteObject(id); };
```

Comment:
So, I think this code could/should use `add_dependent_session` the same way that `Find` does below. That would ensure that the map is correctly cleaned up when the database is closed AND would avoid deleting all of the objects on `reset`. If that implementation doesn't remove db objs from the map after Delete, that's OK for now at least. You can add a follow up work item to do some extra testing and consider fixing that.

Reply:
I have updated `CreateObject` to use `cross-driver-session` and removed `custom_close` on that to avoid deleting object during server reset.

`DeleteObject` still has `custom_close_method` set. That will take care of removing the session from server's map when user invokes `DeleteObject` directly. Does this make sense or am I missing something related to sessions added by `add_dependent_session` can not be cleaned up by normal close methods? I assumed since both initiating session repo and dependent session repos are same (nxDatabaseRef_t), this should work.

Reply:
That sounds and looks correct to me. The test case would be:
1. OpenDatabase
2. CreateDatabaseObj
3. DeleteDatabaseObj
4. CloseDatabase

Step (4) will attempt to close all dependent sessions, including the one that no longer exists. Does it handle that case OK? If not can you change the logic to tolerate the session being there? I don't think that is tested anywhere else, because it's not valid for cross-driver sessions (the driver docs say specifically: don't do it, bad things will happen)

I think it's OK if that's not addressed in this PR. Should be in your acceptance table somewhere.

Reply:
Makes sense to test this scenario out. Added test case for this in acceptance table.